### DISCUSSION TITLE
TypeScript Intellisense not working (npx create-turbo@latest)

### DISCUSSION THREAD

**Discussion by ericchernuka** [vercel/turborepo#295] [Help] [answered]
I'm a bit lost on why this doesn't happen to just work since a friend started a turbo project and got intellisene working flawlessly.

Upon running the CLI to create a new project I went in to see if intellisense just worked (I've had issues in the past where TS just wouldn't find the references). I deleted the import line from the docs index.tsx and tried to have it autocomplete it and it didn't prompt with any suggestions.  I actually went as far as nuking VS Code and all its preferences/settings, but it still does the same thing.  Is anyone able to help me get this working. 🙏🏼 

<img width="404" alt="image" src="https://user-images.githubusercontent.com/243155/146209450-75d7d8a8-e44b-4f50-9b5f-ae6767835f4b.png">

Machine specs:
2021 MacBook Pro M1 Pro
OSX Monteray
VSCode 1.63

**Comment by lionelhorn** [comment]
I encounter similar issue with Intellisense.

I used the `npx create-turbo@latest` template with yarn.

I guess it may be related to the structure of the project and it's monorepo nature. But how to configure IDE's properly?

### From apps/docs
<img src="https://user-images.githubusercontent.com/50528276/146269209-dab62982-6ee8-492f-9336-e81d359f15d5.png" width="400px" />

VSCode:  1.63.1
OS: Windows_NT x64 10.0.19042

Same issue in Webstorm. 

### From packages/ui

On a related note, from the ui package in `Button.tsx` : Webstorm doesn't suggest the proper imports for React.

<img src="https://user-images.githubusercontent.com/50528276/146270355-e892da21-3be3-4c6b-88fc-a2c107f2c963.png" width="400px" />

**Reply by lionelhorn** [reply]
Using the pnpm example from the repo and modifying it with the following instructions https://github.com/vercel/turborepo/discussions/172#discussioncomment-1788065 solves the intellisense issues.

**Reply by ericchernuka** [reply]
@lionelhorn did you find the pnpm works for the Webstorm issue, or both? I tried the pnpm example with the instructions and I still see the same intellisense issues.

**Reply by lionelhorn** [reply]
@ericchernuka  My main IDE is webstorm so I don't know if it's a config issue in vscode. But even with pnpm I still have the issue 

<img src="https://user-images.githubusercontent.com/50528276/146269209-dab62982-6ee8-492f-9336-e81d359f15d5.png" width="400px" />

**Reply by ericchernuka** [reply]
Ya I suspect it's a configuration aspect of TypeScript that is necessary to enable that feature. I realize this isn't a Turborepo issue, but I had hoped that the starter would would magically work and I'd learn how to configure my projects to enable this. Guess most devs are manually importing or have the knowledge of the configuration.

**Comment by NixBiks** [comment]
I think I have a related issue. I do have intellisense in VSCode but it doesn't agree with the configurations from the monorepo - see screenshots:

<img width="489" alt="Screenshot 2021-12-18 at 16 55 05" src="https://user-images.githubusercontent.com/16841792/146647342-69d7d351-1c07-4a1b-b1ff-477b65cdb2da.png">
<img width="545" alt="Screenshot 2021-12-18 at 16 55 21" src="https://user-images.githubusercontent.com/16841792/146647343-d8b7305a-b05c-4eb2-a611-a32d23889bf6.png">


Maybe vscode doesn't support `tsconfig.json` that extends json from installed package? This is my tsconfig (`@dev` is a local package).

```json
{
  "extends": "@dev/tsconfig/nextjs.json",
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "dist",
    "declaration": false,
    "declarationMap": false
  },
  "include": [
    "src",
    "next-env.d.ts"
  ]
}
```

I wonder if @jaredpalmer uses VSCode?

**Reply by jaredpalmer** [reply]
I donuse VSCode. I don't think tsconfig will let you extend rootDir, you have to repeat it in each package or app's tsconfig. Super annoying. Same goes for includes and excludes I believe

**Reply by NixBiks** [reply]
> I donuse VSCode. I don't think tsconfig will let you extend rootDir, you have to repeat it in each package or app's tsconfig. Super annoying. Same goes for includes and excludes I believe 

Sure I can live with that though. But it's worse that vscode throws ts errors but my build script doesn't. Another very odd thing; that specific error above didn't happen when I had yarn v3 but I've downgraded to v1 and now it happens. Very odd indeed

**Reply by NixBiks** [reply]
I finally found the root cause in my case. I had `typescript` as dev dependency in my nextjs app but using a different `typescript` version in my `@dev` package. So removing `typescript` from dev dependency fixed it (instead using the binary from `@dev`)

**Comment by anthonyshew** [comment, marked as answer]
We've updated the examples to have better autocompletion across packages! The latest versions of `npx create-turbo` will have this. For more examples, check out [the examples folder](https://github.com/vercel/turbo/tree/main/examples).

**Reply by vlapo** [reply]
Can you please share with us how to update an existing turborepo project? What was the problem?
